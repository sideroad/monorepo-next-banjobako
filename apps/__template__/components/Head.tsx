import React, { FC } from 'react';
import NextHead from 'next/head';
import config from '../config';

interface Props {
  title?: string;
  description?: string;
  url?: string;
}

const Head: FC<Props> = (props: Props) => (
  <NextHead>
    <meta charSet="UTF-8" />
    <title>{props.title || config.meta.title}</title>
    <meta
      name="description"
      content={props.description || config.meta.description}
    />
    <meta property="og:title" content={config.meta.title} />
    <meta property="og:description" content={config.meta.description} />
    <meta property="og:card" content="summary" />
    <meta property="og:creator" content={config.meta.creator} />
    <meta property="og:site_name" content={props.title || config.meta.title} />
    <meta property="og:url" content={props.url || config.global.url} />
    <meta property="og:title" content={props.title || config.meta.title} />
    <meta property="og:image" content="/static/images/favicon.png" />
    <meta property="og:image:width" content="1200" />
    <meta property="og:image:height" content="630" />
    <meta name="twitter:site" content={props.url || config.global.url} />
    <meta name="twitter:card" content="summary_large_image" />
    <meta name="twitter:image" content="/static/images/favicon.png" />
  </NextHead>
);

export default Head;
