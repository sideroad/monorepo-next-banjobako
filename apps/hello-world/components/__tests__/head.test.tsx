import React from 'react';
import Head from '../head';
import renderer from 'react-test-renderer';

test('Head to render document.head', () => {
  const component = renderer.create(<Head />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
