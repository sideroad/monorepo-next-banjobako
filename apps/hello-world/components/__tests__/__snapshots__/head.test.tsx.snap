// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head to render document.head 1`] = `null`;
